api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceFakeImpl.java

```diff
@@ -0,0 +1,178 @@
+package org.pmiops.workbench.workspaces;
+
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.pmiops.workbench.db.dao.WorkspaceDao;
+import org.pmiops.workbench.db.model.DbUserRecentWorkspace;
+import org.pmiops.workbench.db.model.DbWorkspace;
+import org.pmiops.workbench.exceptions.ForbiddenException;
+import org.pmiops.workbench.exceptions.NotFoundException;
+import org.pmiops.workbench.firecloud.FireCloudService;
+import org.pmiops.workbench.firecloud.model.FirecloudWorkspaceACLUpdate;
+import org.pmiops.workbench.firecloud.model.FirecloudWorkspaceAccessEntry;
+import org.pmiops.workbench.model.UserRole;
+import org.pmiops.workbench.model.WorkspaceAccessLevel;
+import org.pmiops.workbench.model.WorkspaceResponse;
+
+public class WorkspaceServiceFakeImpl implements WorkspaceService {
+
+  @Override
+  public WorkspaceDao getDao() {
+    return null;
+  }
+
+  @Override
+  public Optional<DbWorkspace> findActiveByWorkspaceId(long workspaceId) {
+    return Optional.empty();
+  }
+
+  @Override
+  public FireCloudService getFireCloudService() {
+    return null;
+  }
+
+  @Override
+  public DbWorkspace get(String ns, String firecloudName) {
+    return null;
+  }
+
+  @Override
+  public Optional<DbWorkspace> getByNamespace(String workspaceNamespace) {
+    return Optional.empty();
+  }
+
+  @Override
+  public List<WorkspaceResponse> getWorkspacesAndPublicWorkspaces() {
+    return null;
+  }
+
+  @Override
+  public WorkspaceResponse getWorkspace(String workspaceNamespace) throws NotFoundException {
+    return null;
+  }
+
+  @Override
+  public WorkspaceResponse getWorkspace(String workspaceNamespace, String workspaceId) {
+    return null;
+  }
+
+  @Override
+  public List<WorkspaceResponse> getWorkspaces() {
+    return null;
+  }
+
+  @Override
+  public List<WorkspaceResponse> getPublishedWorkspaces() {
+    return null;
+  }
+
+  @Override
+  public DbWorkspace getRequired(String ns, String firecloudName) {
+    return null;
+  }
+
+  @Override
+  public DbWorkspace getRequiredWithCohorts(String ns, String firecloudName) {
+    return null;
+  }
+
+  @Override
+  public DbWorkspace saveWithLastModified(DbWorkspace workspace) {
+    return null;
+  }
+
+  @Override
+  public void deleteWorkspace(DbWorkspace dbWorkspace) {}
+
+  @Override
+  public void updateWorkspaceBillingAccount(DbWorkspace workspace, String newBillingAccountName) {}
+
+  @Override
+  public void validateActiveBilling(String workspaceNamespace, String workspaceId)
+      throws ForbiddenException {}
+
+  @Override
+  public List<DbWorkspace> findForReview() {
+    return null;
```

Comment:
Why did you decide to send null rather than Empty Object?

Reply:
Oh i see because it was used as new WorkspaceService(null,null...)

Reply:
This is just default codegen in IntelliJ. These should never be called.